src/AgentVisibilityContext.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { createContext } from 'react';
+import { isEqual } from 'underscore';
+import {
+  DisplayAction,
+  ActiveState,
+  VisibilitySelectionMap,
+} from './constants';
+import {
+  getPayloadForHideAll,
+  getSelectionStateInfoPayload,
+} from './selectors';
+import { SelectionStateInfo, UIDisplayData } from '@aics/simularium-viewer';
+
+interface VisibilityContextType {
+  uiDisplayData: UIDisplayData;
+  selectionStateInfo: SelectionStateInfo;
+  allAgentsHiddenState: ActiveState;
+  receiveUIDisplayData: (data: UIDisplayData) => void;
+  updateVisibilityAndSelection: (
+    newValue: VisibilitySelectionMap,
+    hideOrHighlight: DisplayAction
+  ) => void;
+  setAllAgentVisibility: (hiddenState: ActiveState) => void;
+}
+const { Inactive, Active, Indeterminate } = ActiveState;
+
+export const VisibilityContext = createContext<VisibilityContextType>({
+  uiDisplayData: [],
+  selectionStateInfo: {
+    hiddenAgents: [],
+    highlightedAgents: [],
+    colorChange: null,
+  },
+  allAgentsHiddenState: Inactive,
+  receiveUIDisplayData: () => {},
+  updateVisibilityAndSelection: () => {},
+  setAllAgentVisibility: () => {},
+});
+
+export const VisibilityProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const [uiDisplayData, setuiDisplayData] = React.useState<UIDisplayData>([]);
+  const [selectionStateInfo, updateSelectionStateInfo] =
+    React.useState<SelectionStateInfo>({
+      hiddenAgents: [],
+      highlightedAgents: [],
+      colorChange: null,
+    });
+  const [allAgentsHiddenState, setHiddenState] =
+    React.useState<ActiveState>(Inactive);
+
+  const receiveUIDisplayData = (data: UIDisplayData) => {
+    setuiDisplayData(data);
+  };
+
+  const updateVisibilityAndSelection = (
+    newValue: VisibilitySelectionMap
+  ): void => {
+    updateSelectionStateInfo({
+      ...selectionStateInfo,
+      hiddenAgents: getSelectionStateInfoPayload(newValue),
+    });
+    updateAllAgentsHiddenState(newValue);
+  };
+
+  const updateAllAgentsHiddenState = (
+    visibilityMap: VisibilitySelectionMap
+  ) => {
+    let newHiddenState: ActiveState = Indeterminate;
+    if (Object.keys(visibilityMap).length === 0) {
+      newHiddenState = Inactive;
```

Comment:
`Inactive` doesn't seem like the right wording to me. if you want to make it more specific to the check box, I would just have `Checked` and `UnChecked`. If you want to make it more about the state, I would have `AllHidden` and `AllVisible